mne_bids/tsv_handler.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+from collections import OrderedDict
+
+
+def _from_tsv(fname, dtypes=None):
+    """Read a tsv file into an OrderedDict.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file being loaded.
+    dtypes : list
+        List of types to cast the values loaded as. This is specified column by
+        column.
+        Defaults to None. In this case all the data is loaded as strings.
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        Keys are the column names, and values are the column data.
+    """
+    data = np.loadtxt(fname, dtype=str, delimiter='\t')
+    column_names = data[0, :]
+    info = data[1:, :]
+    data_dict = OrderedDict()
+    if isinstance(dtypes, (list, tuple)):
+        if len(dtypes) == info.shape[1]:
+            for i, name in enumerate(column_names):
+                data_dict[name] = info[:, i].astype(dtypes[i]).tolist()
+    else:
+        for i, name in enumerate(column_names):
+            data_dict[name] = info[:, i].tolist()
+    return data_dict
+
+
+def _to_tsv(data, fname):
+    """Write an OrderedDict into a tsv file.
+
+    Parameters
+    ----------
+    data : collections.OrderedDict
+        Ordered dictionary containing data to be written to a tsv file.
+    fname : str
+        Path to the file being written.
+    """
+    n_rows = len(data[list(data.keys())[0]])
+    output = _tsv_to_str(data, n_rows)
+
+    with open(fname, 'w') as f:
+        f.write(output)
+
+
+def _combine(data1, data2, drop_column=None):
+    """Add two OrderedDict's together with optional dropping of repeated data.
+
+    Parameters
+    ----------
+    data1 : collections.OrderedDict
+        Original OrderedDict.
+    data2 : collections.OrderedDict
+        New OrderedDict to be added to the original.
+    drop_column : str
+        Name of the column to check for duplicate values in.
+        Any duplicates found will be dropped from the original data array (ie.
+        most recent value are kept).
+
+    Returns
+    -------
+    data : collections.OrderedDict
+        The new combined data.
+    """
+    new_data = data1.copy()
+    for key, value in data2.items():
+        new_data[key].extend(value)
+    if drop_column in new_data:
+        # get the id of the column to drop repeated values from
+        col_idx = list(new_data.keys()).index(drop_column)
+        # convert to a numpy array and flip vertically
+        arr = np.stack(list(col for col in new_data.values()), axis=1)
```

Comment:
I don't get why you need to do this. Couldn't you just do `np.unique` on the `drop_column` ?